bindings/pydrake/geometry/geometry_py_optimization.cc

```diff
@@ -648,6 +681,14 @@ void DefineIris(py::module m) {
       py::arg("query_object"), py::arg("reference_frame") = std::nullopt,
       doc.MakeIrisObstacles.doc);
 
+  m.def("RayIris",
+      py::overload_cast<const multibody::MultibodyPlant<double>&,
+          const systems::Context<double>&, systems::Context<double>*, const planning::CollisionChecker&, const IrisOptions&, const int>(
+          &RayIris),
+      py::arg("plant"), py::arg("context"), py::arg("mutable_context"), py::arg("checker"), py::arg("options") = IrisOptions(),
```

Comment:
TODO: update interface to only take in a single const context. Note that to do this, we need SceneGraphCollisionChecker to give us not just whether or not a configuration is in collision, but if it is, which pair is in collision. Tommy will open an issue to this effect, and maybe try hacking it in himself.